plugins/events-node/src/api/EventRouter.ts

```diff
@@ -26,13 +24,42 @@ import { EventSubscriber } from './EventSubscriber';
  * @see {@link https://www.enterpriseintegrationpatterns.com/MessageRouter.html | Message Router pattern}.
  * @public
  */
-export abstract class EventRouter implements EventPublisher, EventSubscriber {
-  private eventBroker?: EventBroker;
+export abstract class EventRouter {
+  private readonly eventService: EventService;
+  private readonly topics: string[];
+  private subscribed: boolean = false;
+
+  protected constructor(options: {
+    eventService: EventService;
+    topics: string[];
+  }) {
+    this.eventService = options.eventService;
+    this.topics = options.topics;
+  }
 
   protected abstract determineDestinationTopic(
     params: EventParams,
   ): string | undefined;
 
+  /**
+   * Subscribes itself to the topic(s),
+   * after which events potentially can be received
+   * and processed by {@link EventRouter.onEvent}.
+   */
+  async subscribe(): Promise<void> {
+    if (this.subscribed) {
+      return;
+    }
+
+    this.subscribed = true;
+
+    await this.eventService.subscribe({
+      id: this.constructor.name,
```

Comment:
I think we had a few cases using that. However, it may be problematic as you write 😅 I can change this to a name provided differently.

Also, in case we drop the error/warn log message, we may not even need it at all.